OfflineUX.md

```diff
@@ -0,0 +1,136 @@
+#### Offline UX Patterns
+
+### Contents
+* [Motivation & Philosophy](#motivation-&-philosophy)
+* [UX Pattern Flowchart](#ux-pattern-flowchart)
+* [Answering Questions on the Flow Chart](#answering-questions-on-the-flowchart)
+* [Description of the Patterns](#description-of-the-patterns)
+    - [None - No Offline Behavior](#none---no-offline-behavior)
+    - [A - Optimistic Without Feedback](#a---optimistic-without-feedback)
+    - [B - Optimistic With Feedback](#b---optimistic-with-feedback)
+    - [C - Blocking Form](#c---blocking-form)
+    - [D - Full Page Blocking](#d---full-page-blocking)
+
+### Motivation & Philosophy
+
+Understanding the offline behavior of our app is vital to becoming a productive contributor to the Expensify codebase. Our mission is to support our users in every possible environment, and often our app is used in places where a stable internet connection is not guaranteed. 
+
+The most important concept to keep in mind while reading this document is that we want to allow users to do as much as possible when offline. At first, this might seem impossible because almost everything the user can touch in our app is related to an API request. However, in many cases, we can save that API request and assume it will succeed when the user is back online. We then allow the user to proceed as if their request already succeeded. We call this an optimistic response. Here, we use the word **optimistic** to indicate that we’re confident the request will succeed when the user is online, and we know what that successful response will look like. 
+
+<div style="margin: 15px; padding: 15px; border: 3px solid grey">
+Example: Pinning a chat
+
+When a user clicks the pin button <img style="height: 10px; width: 10px; filter: invert(.5) saturate(5);" src="./assets/images/pin.svg"/> on a chat, two things should happen. 
+
+1. **API Request:** We send a request to the API to keep track of the user's action. After all, if the user signed in on a different device, we would want all of their pinned chats to show up (another way of saying this is we want their pinned chats to ‘persist’ across devices). 
+
+2. **UI Changes:** The chat should go to the top of the list with the other pinned chats, and the pin button should look darker than it did before. This is the user’s visual feedback that clicking the pin button worked.
+
+If the user is offline, do we need to wait for the API request to finish before doing all that visual stuff? No! We are optimistic that the API request will succeed once it’s sent. In the meantime, we let the user continue using the app. 
```

Comment:
I think it's good to ask questions sometimes because the reader will instinctively answer them. If they agree with the answer that's provided in the next sentence, then they know they're comprehending. If they are thinking 'Yes' and reading 'No', that signals that they aren't getting it yet. 

It's a nice technique for pedagogical writing, but I understand if it feels too hand-holdy or patronizing. The edit you gave is restating some of the content from the above paragraph that explains what we mean by *optimistic* UI. 

All in all, I'm not too invested in keeping this as is, so I'm fine with making the suggested edit (with a few tweaks). LMK if you agree or not.

Reply:
I am fine with a revised version of my edits or with a revised version of what you had, as long as both my points above are covered.